src/docquery/ext/pipeline.py

```diff
@@ -417,17 +415,31 @@ def postprocess_extractive_qa(
                 handle_impossible_answer,
                 max_answer_len,
             )
-
             word_ids = output["word_ids"]
             for s, e, score in zip(starts, ends, scores):
-                word_start, word_end = word_ids[s], word_ids[e]
-                if word_start is not None and word_end is not None:
+                if "token_logits" in output:
+                    predicted_token_classes = (
+                        output["token_logits"][
+                            0,
+                            s : e + 1,
+                        ]
+                        .argmax(axis=1)
+                        .cpu()
+                        .numpy()
+                    )
+                    assert np.setdiff1d(predicted_token_classes, [0, 1]).shape == (0,)
+                    token_indices = np.flatnonzero(predicted_token_classes) + s
+                else:
+                    token_indices = range(s, e + 1)
+
+                answer_word_ids = list(unique_everseen([word_ids[i] for i in token_indices]))
+                if len(answer_word_ids) > 0 and answer_word_ids[0] is not None and answer_word_ids[-1] is not None:
                     answers.append(
                         {
                             "score": float(score),
-                            "answer": " ".join(words[word_start : word_end + 1]),
-                            "start": word_start,
-                            "end": word_end,
+                            "answer": " ".join(words[i] for i in answer_word_ids),
+                            "words": answer_word_ids,
```

Comment:
nit: Perhaps this should be `word_ids` or even `answer_word_ids`? I think it's a bit surprising to have the answer `words` be word IDs that's all. (Notably, too, `word_ids` and `words` are the fields on each of the `model_outputs`.)

Reply:
I'm a bit confused -- it is returning `answer_word_ids`. I think the best UX here would be to actually return the `[(word, box)]` (same format as the user input), so that it would work whether or not they ran OCR themselves. But I'd like to do that separately because it will complicate the tests a bit and I want to work out how the demo looks/feels before making another big change.